### PR TITLE
Reset Sequencer state during recovery

### PR DESCRIPTION
**Pull request by zalokhan**
Reset the Sequencer Server state using the FastSMRLoader during the recovery phase.
Steps:
- Server starts, client connects and performs reads and writes.
- Server shuts down/crashes.
- On recovery all the servers start in "not ready state" (PR #762 )
- The management server then seals the servers, runs reconfiguration with the fastSMRLoader and resets the sequencer with the tails.
- After the paxos round, the data path is open to the clients.

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/runtime/clients/SequencerClient.java [hunk 1]
```diff
@@ -3,17 +3,14 @@
...
 import java.util.concurrent.CompletableFuture;
 
 import lombok.Getter;
 import lombok.Setter;
-import org.corfudb.protocols.wireprotocol.CorfuMsgType;
-import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
-import org.corfudb.protocols.wireprotocol.TokenRequest;
-import org.corfudb.protocols.wireprotocol.TokenResponse;
-import org.corfudb.protocols.wireprotocol.TxResolutionInfo;
+import org.corfudb.protocols.wireprotocol.*;
```
Comment (Maithem) [inline runtime/src/main/java/org/corfudb/runtime/clients/SequencerClient.java]: expand

File: test/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java [hunk 2]
```diff
@@ -155,4 +157,56 @@ public void checkBackpointersWork() {
...
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamB))));
+        long tailB = getLastPayloadMessageAs(TokenResponse.class).getToken().getTokenValue();
+
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamC))));
+        sendMessage(new CorfuPayloadMsg<>(CorfuMsgType.TOKEN_REQ,
+                new TokenRequest(1L, Collections.singleton(streamC))));
+
+        long tailC = getLastPayloadMessageAs(TokenResponse.class).getToken().getTokenValue();
```
Comment (Maithem) [inline test/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java]: This just tests the reset part, can you also add tests where the sequencer is restarted and doesn't service token requests until it has been bootstrapped by the management server

File: runtime/src/main/java/org/corfudb/runtime/clients/SequencerClient.java (hunk 1 above)
Comment (zalokhan) [inline runtime/src/main/java/org/corfudb/runtime/clients/SequencerClient.java]: Done

File: test/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java (hunk 2 above)
Comment (zalokhan) [inline test/src/test/java/org/corfudb/infrastructure/SequencerServerTest.java]: Done

**Comment by corfudb-performance** [comment]
Results automatically generated by _CorfuDB Benchmark Framework_ to assess the performance of this pull request for commit **229652c**.

*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Scan workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx+Iterator workload, 10 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 1 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 5 threads, Disk mode
*** 0.0% transaction FAILURE rate for NonConflictingTx workload, 10 threads, Disk mode
 
 An interactive dashboard with Pull Request Performance Metrics for ALL cluster types and numbers of threads in run, is available at:
 [Pull Request #774 Graphs ](https://snapshot.raintank.io/dashboard/snapshot/1HLEyiDfBq8H8idSPnk2yo7M9mGnObfz)

**Pull request merged**